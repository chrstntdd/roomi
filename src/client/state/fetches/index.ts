import Cmd from '@/cmd';

export default store => ({
  signUp: async (state, { email, username, password }) => {
    const mutationName = 'signUp';
    let response;
    try {
      response = await Cmd.mutation(`${mutationName} {
          ${mutationName}(email: "${email}", username: "${username}", password: "${password}") {
            token
          }
        }
      `);
    } catch ({ message }) {
      store.setState({
        graphQlErrorMsg: message
      });
    }

    store.setState({
      jwt: response[mutationName].token
    });
  }
});
