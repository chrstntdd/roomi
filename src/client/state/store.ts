import createStore, { Store } from 'unistore';
import devtools from 'unistore/devtools';

import fetches from './fetches';

export type RootState = {
  readonly email: string;
  readonly firstName: string;
  readonly friends: any[];
  readonly graphQlErrorMsg: string;
  readonly isAuthenticated: boolean;
  readonly jwt: string;
  readonly lastName: string;
  readonly lists: any[];
  readonly role: string;
};

const initialState: RootState = {
  email: '',
  firstName: '',
  friends: [],
  graphQlErrorMsg: '',
  isAuthenticated: false,
  jwt: '',
  lastName: '',
  lists: [],
  role: ''
};

const store: Store<RootState> =
  process.env.NODE_ENV === 'development'
    ? devtools(createStore(initialState))
    : createStore(initialState);

const actions = (store: Store<RootState>) => ({
  ...fetches(store)
});

export { store, actions };
