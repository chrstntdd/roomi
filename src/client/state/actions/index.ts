import { GlobalStore, RootState } from '../store';

export type Sync = {
  setWindowWidth: (state: RootState, { windowWidth: number }) => void;
};

export default (store: GlobalStore): Sync => ({
  setWindowWidth: (_, { windowWidth }) => {
    store.setState({ windowWidth });
  }
});
