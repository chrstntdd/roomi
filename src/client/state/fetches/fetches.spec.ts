import { store } from '@/state/store';
import Cmd from '@/cmd';

import fetches from './';

jest.mock('@/cmd');

const _ = undefined;

test('signUp', async () => {
  const mockToken = '3fc9b689459d738f8c88a3a48aa9e33542016b7a4052e001aaa536fca74813cb';
  Cmd.mutation.mockImplementationOnce(() =>
    Promise.resolve({
      signUp: {
        token: mockToken
      }
    })
  );

  await fetches(store).signUp(_, { email: 'a', username: 'b', password: 'd' });

  expect(Cmd.mutation).toHaveBeenCalledTimes(1);
  expect(Cmd.mutation.mock.calls[0]).toMatchSnapshot();
  expect(store.getState().jwt).toEqual(mockToken);
});
