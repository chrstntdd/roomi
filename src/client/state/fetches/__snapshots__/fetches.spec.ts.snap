// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp 1`] = `
Array [
  "signUp {
          signUp(email: \\"a\\", username: \\"b\\", password: \\"d\\") {
            token
          }
        }
      ",
]
`;
